standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/columnstats/merge/DecimalColumnStatsMergerTest.java

```diff
@@ -180,47 +188,59 @@ public void testMergeNonNullValues() {
 
   @Test
   public void testDecimalCompareEqual() {
-    assertTrue(DECIMAL_3.equals(DECIMAL_3));
+    assertTrue(DECIMAL_3.equals(getDecimal("3", 3, 0)));
+    // the equals method does not check for numerical equality,
+    // e.g., DECIMAL_3 is not equal to getDecimal("3", 30, 1)
   }
 
   @Test
   public void testDecimalCompareDoesntEqual() {
     assertFalse(DECIMAL_3.equals(DECIMAL_5));
+    assertFalse(DECIMAL_3.equals(getDecimal("30", 3, -1)));
   }
 
-  @Test
-  public void testCompareSimple() {
-    DecimalColumnStatsDataInspector data1 = new DecimalColumnStatsDataInspector(DATA_3);
-    DecimalColumnStatsDataInspector data2 = new DecimalColumnStatsDataInspector(DATA_5);
-    assertEquals(DECIMAL_5, merger.mergeHighValue(merger.getHighValue(data1), merger.getHighValue(data2)));
+  private void checkMergedValue(Decimal low, Decimal high) {
+    Objects.requireNonNull(low);
+    Objects.requireNonNull(high);
+    assertTrue(MetaStoreServerUtils.decimalToDouble(low) < MetaStoreServerUtils.decimalToDouble(high));
+    var data1 = new DecimalColumnStatsDataInspector();
+    data1.setLowValue(low);
+    data1.setHighValue(low);
+    var data2 = new DecimalColumnStatsDataInspector();
+    data2.setLowValue(high);
+    data2.setHighValue(high);
+
+    assertEquals(low, merger.mergeLowValue(data1.getLowValue(), data2.getLowValue()));
```

Comment:
IIUC in the end we test the `mergeLowValue` and `mergeHighValue` methods. Both has `Decimal` parameters.
Is it possible to simply call
```
merger.mergeLowValue(low, high))
```
?

Reply:
The usage of `DecimalColumnStatsDataInspector` is covered by the methods that are testing the `DecimalColumnStatsMerger.merge` method. Like 
```
testMergeNullValues
testMergeNullWithNonNullValues
testMergeNonNullWithNullValues
...
```

I think we can skip creating it when testing `mergeLowValue` and `mergeHighValue` and use directly the `Decimal`s `low` and `high`.